synergy-design-system/synergy-design-system
# feat: ✨ Angular: Allow two way data binding component wrappers

# Pull Request

## 📖 Description

This PR provides a change for the angular wrappers that make form values work with two way data binding.

### 🎫 Issues

Closes #397 

## 👩‍💻 Reviewer Notes

- I updated the current angular (dev) dependencies on `@synergy-design-system/angular` and the angular demo to the latest version.
- I opted to move parts of the vue and angular build scripts that where copied over to the `shared.js` build utilities to lessen code duplication.
- After talking to the stakeholder it came apperant that we only need this for the input value fields and __not__ for all attributes.

## 📑 Test Plan

- Check out this branch and build world.
- Move to `packages/_private/angular-demo` and exchange the code for the home component to the snippets defined below.
- Start the angular demo and type text into the input field or change the checkbox or select values. You should see immediate updates appearing.

### Code for `home/home.component.ts`:

```typescript
import { Component } from '@angular/core';

@Component({
  selector: 'home',
  styleUrls: ['./home.styles.css'],
  templateUrl: './home.component.html',
})
export class Home {
  inputValue: string = 'Type something';
  checkboxValue: boolean = false;
  selectValue: string = 'option1';

  setInputValue(value: string) {
    this.inputValue = value;
  }

  log(...args: unknown[]) {
    console.log(...args);
  }
}
```

### Code for `home/home.component.html`:

```html
<h1>
  Welcome to the Synergy Angular Demo!
</h1>

<syn-input [(value)]="inputValue" placeholder="Type something"></syn-input>
<syn-checkbox [(checked)]="checkboxValue">Check me!</syn-checkbox>

<syn-button (click)="setInputValue('Something')">Click me!</syn-button>

<syn-select [(value)]="selectValue">
  <syn-option value="option1">Here</syn-option>
  <syn-option value="option2">there</syn-option>
</syn-select>

<syn-divider></syn-divider>

{{ inputValue }}

{{ checkboxValue }}

{{ selectValue }}

<syn-divider></syn-divider>

```

## ✅ DoD
